Components/Model/Sku.php

```diff
@@ -0,0 +1,172 @@
+ */
+
+use Shopware_Plugins_Frontend_NostoTagging_Components_Helper_Price as PriceHelper;
+use Shopware\Models\Article\Detail as Detail;
+use Nosto\Object\Product\Sku as NostoSku;
+use Shopware\Models\Shop\Shop as Shop;
+use Nosto\Request\Http\HttpRequest as NostoHttpRequest;
+
+class Shopware_Plugins_Frontend_NostoTagging_Components_Model_Sku extends NostoSku
+{
+    protected static $ignoredCustomFieldsProperties = array(
+        'id',
+        'articleDetailId',
+        'articleDetail',
+        'articleId',
+        'article'
+    );
+
+    /**
+     * Loads the SKU Information
+     *
+     * @param Detail $detail Article Detail to load the SKU information
+     * @param Shop|null $shop the shop the product belongs to
+     */
+    public function loadData(Detail $detail, Shop $shop = null)
+    {
+        if (is_null($shop)) {
+            $shop = Shopware()->Shop();
+        }
+
+        $this->setUrl($this->assembleDetailUrl($detail, $shop));
+        $this->setId($detail->getId());
+        $this->setName($detail->getArticle()->getName());
+        $this->setImageUrl($this->getDetailImageUrl($detail));
+        $this->setPrice(floatval(PriceHelper::calcDetailPriceInclTax(
+            $detail,
+            $shop,
+            PriceHelper::PRICE_TYPE_NORMAL
+        )));
+        $this->setListPrice(floatval(PriceHelper::calcDetailPriceInclTax(
+            $detail,
+            $shop,
+            PriceHelper::PRICE_TYPE_LIST
+        )));
+        $this->setAvailable($this->isDetailAvailable($detail));
+        $this->setGtin($detail->getSupplierNumber());
+        $this->setCustomFields($this->getDetailCustomFields($detail));
+    }
+
+    /**
+     * Returns the URL for the given detail
+     * If no image is assigned, will return the parent
+     * article's main image
+     *
+     * @param Detail $detail
+     * @return null|string URL of detail image
+     */
+    protected function getDetailImageUrl(Detail $detail)
+    {
+        $mediaService = Shopware()->Container()->get('shopware_media.media_service');
+        $detailImage = Shopware()
+            ->Models()
+            ->getRepository('Shopware\Models\Article\Image')
+            ->findOneBy(array('articleDetail' => $detail));
+
+        if (!is_null($detailImage)) {
+            $imagePath = $detailImage->getParent()->getMedia()->getPath();
+            return $mediaService->getUrl($imagePath);
+        }
+
+        // Fallback to article main image
+        $articleImgPath = $detail->getArticle()->getImages()->first()->getMedia()->getPath();
+        return $articleImgPath ? $mediaService->getUrl($articleImgPath) : '';
+    }
+
+    /**
+     * Assembles the product url based on article and shop.
+     *
+     * @param Detail $detail the Article Detail model.
+     * @param Shop $shop the shop model.
+     * @return string the url.
+     */
+    protected function assembleDetailUrl(Detail $detail, Shop $shop)
```

Comment:
How about static URL helper that would build the URL? It would take `$store`, `$articleId`, `$detailNumber` (nullable) as parameters. Then SKU builder and product builder could use that

Reply:
Good idea! Fixed, thanks 😃